File: app/src/main/java/com/metahumanz/pacilread/reader/modern/theme/ReaderThemePalette.java
package com.metahumanz.pacilread.reader.modern.theme;

import com.metahumanz.pacilread.R;

public final class ReaderThemePalette {
    public final int backgroundColor;
    public final int pageColor;
    public final int textColor;
    public final int overlayColor;
    public final int backgroundDrawableRes;
    public final String displayName;

    private ReaderThemePalette(int backgroundColor, int pageColor, int textColor, int overlayColor, int backgroundDrawableRes, String displayName) {
        this.backgroundColor = backgroundColor;
        this.pageColor = pageColor;
        this.textColor = textColor;
        this.overlayColor = overlayColor;
        this.backgroundDrawableRes = backgroundDrawableRes;
        this.displayName = displayName;
    }

    public static ReaderThemePalette from(String key) {
        if ("forest".equals(key)) {
            return new ReaderThemePalette(
                    0xFFDCEAD7,
                    0xFFEAF4E6,
                    0xFF2A4B2A,
                    0xB8EEF6E9,
                    R.drawable.theme_bg_forest,
                    "护眼"
            );
        }
        if ("night".equals(key)) {
            return new ReaderThemePalette(
                    0xFF0B1320,
                    0xFF121C2B,
                    0xFFDDE7EE,
                    0xD20B121D,
                    R.drawable.theme_bg_night,
                    "夜航"
            );
        }
        return new ReaderThemePalette(
                0xFFF4ECD8,
                0xFFF7F0E1,
                0xFF5C4B37,
                0xA6FFF8ED,
                R.drawable.theme_bg_paper,
                "纸控"
        );
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/reader/modern/theme/ReaderThemePalette.kt
package com.metahumanz.pacilread.reader.modern.theme

import com.metahumanz.pacilread.R

class ReaderThemePalette private constructor(
    @JvmField val backgroundColor: Int,
    @JvmField val pageColor: Int,
    @JvmField val textColor: Int,
    @JvmField val overlayColor: Int,
    @JvmField val backgroundDrawableRes: Int,
    @JvmField val displayName: String,
) {
    companion object {
        @JvmStatic
        fun from(key: String?): ReaderThemePalette {
            if (key == "forest") {
                return ReaderThemePalette(
                    0xFFDCEAD7.toInt(),
                    0xFFEAF4E6.toInt(),
                    0xFF2A4B2A.toInt(),
                    0xB8EEF6E9.toInt(),
                    R.drawable.theme_bg_forest,
                    "护眼",
                )
            }
            if (key == "night") {
                return ReaderThemePalette(
                    0xFF0B1320.toInt(),
                    0xFF121C2B.toInt(),
                    0xFFDDE7EE.toInt(),
                    0xD20B121D.toInt(),
                    R.drawable.theme_bg_night,
                    "夜航",
                )
            }
            return ReaderThemePalette(
                0xFFF4ECD8.toInt(),
                0xFFF7F0E1.toInt(),
                0xFF5C4B37.toInt(),
                0xA6FFF8ED.toInt(),
                R.drawable.theme_bg_paper,
                "纸控",
            )
        }
    }
}
